zipkin-storage/cassandra/src/main/java/zipkin/storage/cassandra/CassandraSpanConsumer.java

```diff
@@ -136,12 +141,24 @@ private RegularStatement maybeUseTtl(Insert value) {
     return transform(Futures.allAsList(futures.build()), TO_VOID);
   }
 
+  private static boolean isServerRecvSpan(Span span) {
+    for (int i = 0, length = span.annotations.size(); i < length; i++) {
+      Annotation annotation = span.annotations.get(i);
+      if (annotation.value.equals(Constants.SERVER_RECV)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
   /**
    * Store the span in the underlying storage for later retrieval.
    */
-  ListenableFuture<?> storeSpan(long traceId, long timestamp, String key, ByteBuffer span) {
+  ListenableFuture<?> storeSpan(long traceId, long timestamp, boolean isServerRecvSpan, String key, ByteBuffer span) {
     try {
-      if (0 == timestamp && metadata.compactionClass.contains("DateTieredCompactionStrategy")) {
```

Comment:
It would need to pass the span though and from what I see in the code, this is not what was intended (we extract everything needed from the span before)